central/scannerdefinitions/handler/handler.go

```diff
@@ -261,7 +261,7 @@ func (h *httpHandler) post(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if err := h.handleZipContentsFromVulnDump(r.Context(), tempFile); err != nil {
-		httputil.WriteGRPCStyleError(w, codes.Internal, err)
```

Comment:
Why this change?

Reply:
Previousy, `codes.Internal` lead to a 500 HTTP status code, which the retryable client will try to retry on (since per definition, this doesn't really mean that anything was _wrong_ with the provided data but rather a server issue).

While we could adjust the retry policy, this felt like a wrong status code on the backend side: we return an error because the provided file wasn't of ZIP format, so invalid argument was a more appropriate status code to return.

Reply:
Are you sure the retry handler would retry a POST by default? It's not guaranteed to be idempotent. But I get your point, it wasn't obvious to me that the error here is user induced.

Reply:
Default behavior of the client is to retry irrespective of the HTTP method.

We may adjust this going forward, but for the time being should be OK.